### PR TITLE
Feature/multi classification

### PR DESCRIPTION
**Pull request by AngranLi**
The metric that converts the prediction from multi-class to binary (for MSG), and calculate the AUC.

Change type:

- [ ] Bug fix
- [x] Feature
- [ ] Documentation

Checklist:

- [x] My code follows the style guidelines of this [project](https://docs.google.com/document/d/1jckZJe0CrWyF-IjxoO2OfBKAyKLC3Yk9Xr7UmOLBETA/edit?usp=sharing)
- [x] I have performed a self-review of my own code
- [x] I have commented my code and added docstrings to all exposed functions and classes
- [x] I have made corresponding changes to the documentation
- [x] Any relevant changes to third party licenses have been updated in the README

### CODE REVIEW

**Comment by jwmandeville** [comment]
<del>I just tested this out, and for some reason it's crashing the kernel in my jupyter notebook. It happens when I call `functional.train`. This is the output in the console: 
</del>

Turns out this is an unrelated issue, ignore this comment

**Comment by jwmandeville** [comment]
Tested and it's working with the Audio example notebook

File: xt_training/metrics.py [hunk 1]
```diff
@@ -247,6 +247,21 @@ def reset(self):
...
+class MultitoBinaryAUC():
+    """Convert the prediction from multi-class to binary, and calculate the AUC."""
+    def __init__(self):
+        self.roc_auc = ROC_AUC()
+
+    def __call__(self, y_pred, y):
+        y_pred_bi = torch.transpose(
+                        torch.vstack([y_pred[:, 0], torch.logical_not(y_pred[:, 0])]), 
+                        0, 1
+                    )
```
Comment (timesler) [inline xt_training/metrics.py]: `y_pred` is generally logits. As a result, shouldn't `torch.logical_not(y_pred[:, 0])` always return False for every row?

I would have thought that correct way to do this was to just do:

```python
y_pred = nn.functional.softmax(y_pred, dim=1)
y_pred_bi = torch.vstack(y_pred[:, 0], y_pred[:, 0:].sum(dim=1)).log()

y_bi = (y > 0).float()
```

**Review by timesler — CHANGES_REQUESTED** [review]
More generally, I can't think of any general applications of this metric. The use case where you want to treat all non-zero labels as a single class had a specific use in MSG, but not in general. However, if we add an argument to the constructor that allows the user to specifiy which class to keep (i.e., the class we _don't_ want to combine), then this might be useful

File: xt_training/metrics.py (hunk 1 above)
Comment (AngranLi) [reply xt_training/metrics.py]: Thanks Tim! This was a stupid mistake. `y_pred` should be logits, and we need to make sure `y_pred_bi` is logits as the input of `ROC_AUC()`. Hence I changed the code following your advice:
```
y_pred = F.softmax(y_pred, dim=1)
y_pred_bi = torch.transpose(
    torch.vstack([y_pred[:, 0], y_pred[:, 1:].sum(dim=1)]).log(),
    0, 1
)
y_bi = (y > 0).float()
```

**Comment by AngranLi** [comment]
For the generality, I think it might be useful to let the user choose which classes they want to combine as the "zero class", and the remaining classes would be regarded as the "one class". Hence I added an argument `zero_cls` to the constructor.

**Review by timesler — APPROVED** [review]

**Comment by timesler** [comment]
Looks great!
